src/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -649,8 +655,9 @@ private void shiftIndicesLeft(int shiftAmount) {
      * Fills the buffer with up to the requested number of additional bytes. It is the caller's responsibility to
      * ensure that there is space in the buffer.
      * @param numberOfBytesToFill the number of additional bytes to attempt to add to the buffer.
+     * @return the number of bytes filled.
      */
-    private void refill(long numberOfBytesToFill) {
+    private int refill(long numberOfBytesToFill) {
```

Comment:
I realize this is a private method and only used in the `fillAt` method, for now. I would put the loop here lest we add some other method that calls this and have the same problem. Or being crystal clear in the javadoc that it is the caller's responsbility to retry, though then we'd have two callers doing looping that _could_ be here (unless I'm missing something).

Reply:
Good point, moved the loop into `refill`. Also added a test to ensure skipping over a value works when the InputStream has this behavior.